### PR TITLE
Complete overhaul of xenos larva generation

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Replace the curent game mechanic of killing marines --> headbiting --> siloing with a passive larva generation by a new structure, the spawing pool. Remove everything related to the old mechanic, including headbiting, voring dead bodies, grabing dead bodies, siloing, silos, free larva when marines are spawning or respawning (actually this mechanic stay the same until shutters open, so population is still balanced before the fight begin).

Spawning pool are generated in multiple possible locations in caves, this position being already fortified at start (thick walls and doors, eggs). The number generated is directly linked to the number of players, as to make the game as balance in low pop than high pop. 

One spawning pool will generate 1/16 of a larva every minute. For exemple, in high pop there will be 4 pools at start, so the hive will have a new larva every 4 minutes. You can create an unlimited number of pools. Process only start after shutters open

Psychic points are back, queen or shrike can both corrupt generators that will then generate such points. One spawning pool will cost 900 points, so that's 30 minutes with 10 generators online (every map has exactly 10 generators). Marines can remove the corruption using a welder.

Killing all the pools is another winning conditon for marines, as it will give 10 minutes to the hive to build another one, or it's game over.

The hive get a free spawning pool when hijacking. This pool will generate larva depending on the number of marines, currently it's set a 1 larva every minute if you have 40 marines. Destructing that pool is not a victory condition shipside

Add more caves to ice colony, so it has more space for spawning pools at start of the game

Move green nuke generator in magmoor, so it's no longer right next to a spawning pool.

## Why It's Good For The Game

Here are the goals of this PR:

- Deleting the FOB meta. Marines now have zero incentive to stay in fob, as their ennemy will only grow stronger and stronger. 
- Marines have clear objectives to kill: the pools. This will make organising them probably easier, and it removes any possiblity for the xenos to delay
- Remove the salt of headbitings, or getting sent to brazil. Xenos now have no capabilities to perma kill you, or even to take you away from your comrades. That means that if the medic aren't complete idiots, you will recover most of the time
- Making xenos not afraid to attack FOB again and again, even if that means dying often. As they have a supply of larvas at home, they don't need to be that efficient in their fights. That'll make fob sieging get short.
- Give back a purpose to pods and tapole. Flanking to kill pools will be a valuable strat
- Alleviate the downside of having retard xenos. Currently, if 5 stupid xenos die, it's a disaster, and will make the rest of the round almost already over. With this, losing xenos is bad, but far less insurmountable as you only need to survive to get those larvas back
- Make xenos and marines fight in the middle and the map for generator control

Overall, we switch the focus off the game toward ground control, rather than death and attrition battle

The reason I'm removing all the body dragging stuff is for one reason: death is now way less impactfull as a xenos, you can recover easily with a 2 minutes downtime. This is unfair for marine, even if we just remove headbite and siloing, nothing prevent xenos to just vore or drag boddies away and make it a permadeath. I believe that without removing body drag, this PR will only cause a huge tank of marine WR

Of course, this will probably tank the WR in one way or another, as every number is more or less put arbitarly by me and only me. This will probably need a lot of tweaking and changes, and it won't be perfect the first time. 

Be sure to give me feedback in this PR, or tag me in the discord feedback channel (MrSnoopy)

It also breaks crash balance in the gutter, but fuck crash anyway.

## Changelog
:cl:
add: Spawning pools. They are randomly generated in caves, and each one will generate 0.5 larva points every minute. You need 8 points for a new larva. The number of spawning pool generated at start depends of the number of players, and they only spawn larva after shutters are opened
add: Psychic points. Queen and shrike can corrupt generators (10 in each map), marines can remove that corruption using a welder
add: Spawning pools can be constructed using psych points (900 points, wich corresponds to 30 minutes with 10 generators). 
You are given points to build pools at starts, depending on the number of players on the server
Special pool can be built when hijack, the spawn rate of this upgraded pool is linear with the number of marines alive. Psych points are resets to 900 when hijacking, so you only get one and the other are destroyed when launching alamo
balance: Marines have a new objective: if they kill all the pools, xenos have ten minutes to build another one or they lose.
del: Headbiting is gone
del: Voring is gone
del: Silos are gone, you can't silo bodies to gain larva
tweak: Passive larva generation when marines are joining the game only works when shutters are closed, after that spawning pools are starting
tweak: Xenos can't grab dead humans anymore, only living ones
imageadd: Spawning pool sprite, stole from cimex sprite
/:cl:

### CODE REVIEW

**Comment by RuneQuester** [comment]
POg

**Comment by Swagile** [comment]
this truly be the based xeno update that we needed, but didn't know we wanted

plus gives back a reason for corpsmen to exist besides slapping people with meraderm and IA occasionally

the problem i foresee here is that there will be lots of burrowed but not a lot of xenos, so why not tie xeno evolution to the spawning pools? aka the more pools you have, the faster you evolve. but start this process at 12:33 so as to avoid ancient everything the moment marines land.

this is balanced around the fact that even if you have a lot of spawning pools, you still have to DEFEND the pools, so random drop podders or chadpole drops can fuck up your investment on pools

**Comment by ilikesaying** [comment]
pretty sure burrowed larva increase the upgrade and evolve point generation, with 5 larvae causing xenos to evolve and upgrade twice as fast

**Comment by davidchan13** [comment]
:+1: for removing vore. TM ASAP please.

**Comment by Al-1ce** [comment]
defibbing will be braindead easy without the ability to drag bodies but the rest seems good
based pr

**Comment by SplinterGP** [comment]
>Voring is gone
How dare you, feeble man

**Comment by TG-Eris** [comment]
I really like the idea of PR#5800, though I'm concerned that if anything marine win rates are going to increase as a result.  Removing dragging alone will make you beloved amongst all Marine players, it's an annoying mechanic overall and removing it allows Marines to play more aggressively.  Overall this PR should increase aggression on all sides, which most people would probably agree is a net positive.  I did have a couple of suggestions that could be terrible or decent, at the very least, food for thought:

-Not capping the amount of spawning pools will probably result in them being pushed out as soon as psych points are available in as safe a place as possible.  Even with them not providing free larva generation after a certain point, they can be used as a buffer for Marines destroying a spawning pool, making the mechanic in which marines hunt them down less likely to be undertaken.  I would also consider reducing the cost of producing a pool from 30 minutes with full generators to 20, and setting a cap on the total psych points Xenos can accumulate so that they don't have the ability to suddenly spam out pools in various parts of the map towards the end of the round simply to delay.
-Adding an ability to one of the CIC consoles that allows for seeing the approximate locations of the spawning pools (say a 25x25 square) so that Xenos have to actively defend the area and not just rely on Marines never entering the caves until the round is over.  This would further your goal of providing additional utility for tadpole's and drop pods, while also giving marines a concrete waypoint to work towards.  Right now marines just kind of push in a direction aimlessly, this could help with that.

**Comment by Walarks** [comment]
Vore should be a config or being made unusable instead of outright removing it from code imo.

**Comment by ondrej008** [comment]
no love for crash? 😢

**Comment by BraveMole** [comment]
> -Not capping the amount of spawning pools will probably result in them being pushed out as soon as psych points are available in as safe a place as possible. Even with them not providing free larva generation after a certain point, they can be used as a buffer for Marines destroying a spawning pool, making the mechanic in which marines hunt them down less likely to be undertaken. I would also consider reducing the cost of producing a pool from 30 minutes with full generators to 20, and setting a cap on the total psych points Xenos can accumulate so that they don't have the ability to suddenly spam out pools in various parts of the map towards the end of the round simply to delay.

Yeah, actually caping both can be a good idea

> -Adding an ability to one of the CIC consoles that allows for seeing the approximate locations of the spawning pools (say a 25x25 square) so that Xenos have to actively defend the area and not just rely on Marines never entering the caves until the round is over. This would further your goal of providing additional utility for tadpole's and drop pods, while also giving marines a concrete waypoint to work towards. Right now marines just kind of push in a direction aimlessly, this could help with that.

I had the idea of using the nuke disk tracker for that, but that'll be done after

**Comment by AyyRobotics** [comment]
I like the ideas behind this, though I think there's a few issues. 
**First**, if xenos are meant to be defending these all the time, then that could only result in either having to stretch xenos thin having to defend them all and fight the marines, having a runner or some other fast xeno to patrol them, or just fortifying one position and putting all the pools there, effectively reversing the FOB meta. 
**Second**, the tadpole. Assuming marines find out where the naturally spawned pools positions are, what's to stop a tadpole rush? If the xenos are all huddled around a giant Xenofob of pools then it'd be useless, but with naturally spawning ones, and ones that might not even have traps, going by what I previously said the only means of defense is either to spread forces very thinly, or drop everything to rush the pool when the scouting runner screams over hivemind. 
And **third**, you can't even do that because of generators, There's just even more things that xenos need to defend, while also fighting the main force, and tadpole, assuming it's being used for mining and not flanking.

**TL;DR**: It's a good idea but it looks like it may force xenos to scramble to defend every single spawnpool, and generator, and fight the main force, and fight the tadpole. Might even reverse the FOB meta this aims to stop.

**Comment by BraveMole** [comment]
> **TL;DR**: It's a good idea but it looks like it may force xenos to scramble to defend every single spawnpool, and generator, and fight the main force, and fight the tadpole. Might even reverse the FOB meta this aims to stop.

Yeah, it's possible it will be shit.

**Comment by Tennessee116** [comment]
We finally cranking out the good game design and putting the psychic points to use.

**Comment by SilencedMP5A5** [comment]
based

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -451,7 +449,7 @@
 #define COMSIG_XENOABILITY_DROP_WEEDS "xenoability_drop_weeds"
 #define COMSIG_XENOABILITY_CHOOSE_RESIN "xenoability_choose_resin"
 #define COMSIG_XENOABILITY_SECRETE_RESIN "xenoability_secrete_resin"
-#define COMSIG_XENOABILITY_SECRETE_RESIN_SILO "xenoability_secrete_resin_silo"
+#define COMSIG_XENOABILITY_SECRETE_RESIN_SPAWNING_POOL "xenoability_secrete_resin_spawning_pool"
```
Comment (TiviPlus) [inline code/__DEFINES/dcs/signals.dm:452]: Dont change the string here or it will break stuff when we TM

File: code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm [hunk 2]
```diff
@@ -707,3 +705,30 @@
 	SSblackbox.record_feedback("tally", "round_statistics", -1, "total_xenos_created")
 	qdel(T)
 	X.use_plasma(600)
+
+/datum/action/xeno_action/activable/corrupt_generator
+	name = "Corrupt generator"
+	action_icon_state = "tunnel"
+	mechanics_text = "Corrupt a generator to begin increasing the psycic energy of the hive."
+	plasma_cost = 200
+	keybind_signal = COMSIG_XENOABILITY_CORRUPT_GENERATOR
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm:714]: comment this ou or it will runtime when we tm

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/hive_datum.dm [hunk 3]
```diff
@@ -1114,3 +1114,25 @@ to_chat will check for valid clients itself already so no need to double check f
...
+	if(GLOB.xeno_resin_spawning_pools.len)
+		if(D?.spawning_poolless_hive_timer)
+			deltimer(D.spawning_poolless_hive_timer)
+			D.spawning_poolless_hive_timer = null
+		return
+
+	if(D?.spawning_poolless_hive_timer)
+		return
+
+	message_admins("Silo less collapse started")
```
[the remark this answers is not in the document]
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/hive_datum.dm:1136]: debug, i'm removing

**Review by TiviPlus — APPROVED** [review]
whee

**Comment by SplinterGP** [comment]
tbh i'd like this as another gamemode, maybe distress variation, Distress(TDM), Distress(Objective), Crash, something like that, just mostly because i think giving people option to vote what type they wanna play is good and keeping people from burning out or getting bored from same gameplay loop, giving them some difference on gameplay once in a while, mostly my opinion, can be done later after testing or something as well.

**Comment by SplinterGP** [comment]
>Deleting the FOB meta. Marines now have zero incentive to stay in fob, as their ennemy will only grow stronger and stronger.

depends as far as it goes, if theyre killing xenos that are FOBCamping/sieging they aint getting any stronger.

**Comment by MiniMeatwad** [comment]
This as a gamemode variation much like crash and distress is instead of changing distress' gameplay loop would be cool.

**Comment by Al-1ce** [comment]
> tbh i'd like this as another gamemode, maybe distress variation, Distress(TDM), Distress(Objective),

Gotta agree with this.

**Comment by Drevenshek** [comment]
I really like this, marines can now play to kill or go for objectives which is based. Voring and headbiting going makes the game more fun for marines while xenos can play more aggressively and freely as death doesen't matter as much, this is a really good solution for completely ridding capture gameplay

**Comment by Drevenshek** [comment]
Also make it so xenos can't drag anyone, not crits or dead. Now xenos will just keep people alive they think are dangerous and drag them to kingdom come, which just feels dirty and kind of against what xenomorphs should be, alien killing machines.

**Comment by SilencedMP5A5** [comment]
xenos will drag people away no matter what, just kill dragging entirely

**Comment by Drevenshek** [comment]
yeah theirs no reason for it to be anymore, capture gameplay should be gone for GOOD

**Comment by XSlayer300** [comment]
Picking "cancel" when you want to spawn as a larva sends you to nullspace and adds this:
image

**Comment by TiviPlus** [comment]
> Picking "cancel" when you want to spawn as a larva sends you to nullspace and adds this:
> --image--

this is probably that bug hugh caused

**Comment by TiviPlus** [comment]
On the game mode side, distress is shit and I'd rather replace it with a functional thing such as this 
If someone wants to do "old distress gamemode" you'll have to justify it in a different PR

**Comment by BraveMole** [comment]
Small changes:

Fixed the typo, now you really have to wait 30 minutes to get a new pool and not 300.

When you hijack, you get a free super pool that will spawn larva much faster

**Comment by SplinterGP** [comment]
> On the game mode side, distress is shit and I'd rather replace it with a functional thing such as this
> If someone wants to do "old distress gamemode" you'll have to justify it in a different PR

i won't question that distress is bad, because that's opinion, and it depends of how this goes, it can in the end not be much different than crash but in distress setting instead of in canterbury, remains to be seen

**Review by TiviPlus — APPROVED** [review]

**Comment by BraveMole** [comment]
Larva generation is buffed a little, to prevent stalemates. This will force marines to PUSH, and not stay near FOB. Use pods or tadpole, or die

**Comment by SplinterGP** [comment]
> Larva generation is buffed a little, to prevent stalemates. This will force marines to PUSH, and not stay near FOB. Use pods or tadpole, or die

Tadpole and Pods are transport tools not push tools, kek.

**Comment by TG-Eris** [comment]
So in its current state after being in several rounds with this PR TM'd I think we've got some problems.

As it is, as long as Xenos keep up maximum pressure on FOB without committing themselves to secure kills, (which they don't need to do) and playing fairly safe they can prevent Marines from ever being able to push out of FOB while passively getting larva in the background.  Unless Marines get an early advantage by wiping out large amounts of Xenos the round is up to the Xenos to lose.  Marines only have the power to push caves when Xenos are very weak, so it's not really a viable option for most of the round.  Rounds last ages because Marines can't leave FOB and Xenos can't lose unless they throw a massive number of sisters away.  This wasn't an issue in the past, as a single Xeno loss was devastating if the Xenos weren't actively recovering bodies, meaning that if they ineffectually sieged FOB for any length of time they ran the risk of being attritioned down. 

Another change which is in theory a buff for Marines, the removal of dragging dead bodies, is in practice a nerf.  Competent Xeno players now understand that if they don't remove the marines from the battlefield while they're alive the chances of the body being recovered during skirmishing combat is very high, if not guaranteed.  Now Xenos  combat drag more than ever, as it's necessary to secure Marine kills.  Beforehand Xenos would go for kills, or attempt to drive off other Marines and potentially get more kills, as they knew that as long as the body didn't go anywhere they could drag it back once they were safe.

As Splinter stated, pods are actually suicide as a flanking tool, unless your plan is to flank in very close proximity to a currently ongoing Marine push, because if you don't get Marine support immediately you are going to be wasted.  Remember, Xenos have perfect vision, are significantly faster and heal more easily than Marines.  They won't be able to outflank Xenos most of the time and if they pod into a corner of the map they'll get spotted by hivemind or another Xeno and swarmed before they can even get to the pool.  If they somehow do destroy the pool, they're still going to get completely wiped out afterwards unless, like I stated earlier, Xenos are basically defeated and Marines are pushing caves.  The Tadpole has a very long cooldown, now that Xenos have become more familiar with it and it can't go in caves not only is it not the terror it used to be, but any time it's used as a miner tool or flanking tool it gets wiped out.  If you drop the Tadpole near cave entrances one of two choices becomes available to you.  You can, A. Push into caves, leaving the tadpole undefended where it will very soon be swarmed and overrun, cutting off your escape route.  Maybe you get to the Spawning Pool, maybe you don't, but you're going to die unless Xenos are already basically wiped out.  Or B. Camp at Tadpole and build up a minifob, which is _going_ to be overrun and doesn't help you take out the spawning pools.  So, we're back to needing the main marine force from FOB needing to come deal with a part of the map they don't see unless the match is basically won most rounds.

I was very excited for this TM, as I expected it to increase the aggression level of both parties, but that's because I misunderstood  the nature of the dragging changes to mean that Xenos couldn't drag Marines at all.  Marines are now _required_ to push to have any chance at winning, but pushing is harder than it was before due to Xenos being able to afford more aggressive play and Marines getting dragged from combat more frequently.

So, instead of just complaining, what can be done besides scrapping the whole thing:
**On Dragging**
-Reduce the speed significantly by which Xenos can drag living Marines.  This means that Marines can't be zipped straight out of an engagement, and Xenos will need to take an actual risk to pull them back.
-Alternatively, or in addition, make it to where Xenos can't melee a target they're dragging.  If you've been grabbed you're probably shafted, as you're probably already in a bad state and the constant incoming damage from the Xeno dragging you makes it impossible to recover.
-Another one! Make Xenos windup to begin dragging, so they go to drag and it takes 1.5 seconds or so before they actually have a grasp on the Marine.
**On Spawning Pools**
-I'd probably reduce the rate at which Xenos get Larva by a small amount.  Xenos now have burrowed larva almost constantly throughout the round, with their principal limiter being not having enough players.  I realize that this is an issue not entirely related to this PR, Xenos frequently have larva with no players, but now it's much more prevalent. In the last few rounds when a Xeno player dies, he simply has to wait the Xeno timer out and he's immediately back in, even in the rounds where Xenos are performing poorly.
-The spawning pool locations need to either be standardized, so that they're always in the same exact location, or Marines absolutely need to know exactly where they are.  I realize the walls are a different color and they glow, but I mean Marines need to know what direction the spawning pool is from across the map.
-Have it connected to the amount of Marines active.  Enjoy coding that nightmare in, but I doubt one size fits all is going to work for all pop amounts.

I don't have any other suggestions really.  Hopefully some of this is useful.  As it is I think this is going to be very difficult to balance properly.

**Comment by SpaceLoveSs13** [comment]
As eris mentioned before, I have made sure more people give their feedback to this TM. So here's mine from an admin perspective.

In short, it is stagnating round by a lot.

First issue I have noticed is how fast it can generate larvas so any xenomorphs who die can instantly respawn after 2 minutes. An endless wave of xenomorphs. 

Second issue, xenoes not being able to headbite means marines can just fobbitcamp and retrieve bodies every once in a while their squad members die. We just had a round and I observed entire round and I can assure you it is happening a lot.

Third thing is that xenomorphs constantly coming back make marines not being able to push(specially in high pop) so marines can't do their objectives such as secure miners go destroy corruption or pools to stop this from happening either.

These are all things that I myself saw in a round. I believe, this pr at it's current situation should be tweaked heaviliy. Rounds being stagnated to 2-3 hours is not fun for neither sides or ghosts. I hope these issues get adressed soon.

**Comment by BraveMole** [comment]
So minor tweaks:

- If you have the gene, you can now make a pool every 15 minutes instead of 30
- Reduce the larva generation by pools from 0.5 to 0.3
- Remove the limit of pool you can build

So marines have a clear shot to push at the begining, since larva generation will be slower. But, if they fail to push, the pools will starts to stacks and xenos will finish by overruning the marines

**Comment by Swagile** [comment]
I would like to state my opinion as someone who has played both sides, both groundside marine, RO / CIC, and as a Xeno.

So far this update is great. It makes squad engis and corpsmen relevant again (since headbite is gone and engis need to build forward cades again since marines actually have to push out of FOB). Tadpole is back in the meta if sent to the front near caves as it gives marines an evac option after killing one or two pools and retreating with wounded.

The reason rounds are long right now is because marines are NOT ADAPTING; they think killing 3-5 xenos matters still. It does not; xenos can come back now even with 1 or 2 T3 deaths as long as entire hive does not die in a matter of 10 minutes. A lot of the time marines aren't capitalizing on the ground they gain and cading areas, nor are they using tadpole, so they get stuck in FOB and die because they have no miners and they keep killing T1's and T2's which just comes back.

Once marines minds shifts from "me kill xeno, me good player" to "me play strategically, me kill pool and fall back", marines will start winning a lot more.

For example, we had a round where I was RO and it lasted 3 hours; I sent down 10 + B17 sets, MULTIPLE miniguns, SADAR, IFF sniper kits, and yet the marines were STILL in a stalemate. It took the sacrifice of Delta SL Steve Phobos and 3 other volunteers (who all died in this mission) to rush Lambda caves and kill as many pools as they can. They killed 4, which gave the marines the ability to finally push and wipe out the xenos 30 or so minutes later.

On the xeno side, being a T1 and T2 are no longer shit. You can afford to do risky plays because you know you can come back 2-3 minutes later, which means swarm tactics and actually punishing marine stupidity is viable even on OPEN GROUND (this is important, since marines like to camp open grounds a lot with TL bolters and GPMG's). This changes the gameplay loop for competent xenos; they can afford losses and not snowball into a complete loss simply because they lost a few xenos and no longer have to play perfectly and rely on marine stupidity as much. They can do ballsy plays and get away with it, which makes xeno enjoyable.

There is also complaints that the rounds are longer now (2 hours to 3 hours average) but thats completely fine in my opinion because these aren't just boring rounds where you camp FOB forever; its constant back and forth of marines trying to gain miners or push past FOB, and xenos trying to pull back to caves to recover numbers then go back out to harass marines. Its no longer a "do or die" 5-10 minute fight where the entire game is decided in one big ungaball because xenos cannot afford even a single loss.

Overall, while there are tweaks needed to be added (like a pinpointer or some way to actually FIND pools besides aimlessly wandering caves), this is a PR that positively affects gameplay.

**Comment by Losenis** [comment]
What swagile said. I didn't know how much better pools were than silos till the pr got unTMd. Also, how will the psychic generation work with maps that have differing number of gens compared to others? Like Magmoor vs Big Red. Is it compensated somehow to generate the same number of points total or will xenos just be weaker in some maps?

**Comment by BraveMole** [comment]
> What swagile said. I didn't know how much better pools were than silos till the pr got unTMd. Also, how will the psychic generation work with maps that have differing number of gens compared to others? Like Magmoor vs Big Red. Is it compensated somehow to generate the same number of points total or will xenos just be weaker in some maps?

Yes, it's compensating.

**Comment by spookydonut** [comment]
This contains a massive amount of changes that should be split up.

At a minimum the map changes should be in a separate PR.

There is also a LOT of code churn with needless renames and poor maintainability code.

**Comment by TiviPlus** [comment]
Map changes removed

**Review by TiviPlus — APPROVED** [review]
